rotkehlchen/tests/unit/decoders/test_balancer.py

```diff
@@ -331,3 +332,310 @@ def test_balancer_trade(ethereum_inquirer, ethereum_accounts):
+        chain_id=ChainID.ETHEREUM,
+        token_kind=EvmTokenKind.ERC20,
+        symbol='BCoW-80QQQ-20WETH',
+        name='Balancer CoW AMM 80 QQQ 20 WETH',
+        decimals=18,
+        protocol=CPT_BALANCER_V1,
+        underlying_tokens=[
+            UnderlyingToken(  # including just one of two underlying token.
+                address=string_to_evm_address('0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'),
+                token_kind=EvmTokenKind.ERC20,
+                weight=FVal('0.2'),
+            ),
+        ],
+    )
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=ethereum_inquirer, tx_hash=tx_hash)
+    user_address, timestamp, gas_str, approve_amt, receive_amt, deposit_qqq_amt, deposit_weth_amt = ethereum_accounts[0], TimestampMS(1730837051000), '0.002106711205202376', '0.000000001', '4152.559258169060848717', '3545739.463723403', '0.726672467956343618'  # noqa: E501
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_str)),
+            location_label=user_address,
+            notes=f'Burn {gas_str} ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=164,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.APPROVE,
+            asset=Asset('eip155:1/erc20:0x690031313d70c2545357F4487C6a3F134C434507'),
+            balance=Balance(amount=FVal(approve_amt)),
+            location_label=user_address,
+            notes=f'Set QQQ spending approval of {user_address} by {balancer_qqq_weth_pool_token.evm_address} to {approve_amt}',  # noqa: E501
+            address=balancer_qqq_weth_pool_token.evm_address,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=165,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=balancer_qqq_weth_pool_token,
+            balance=Balance(amount=FVal(receive_amt)),
+            location_label=user_address,
+            counterparty=CPT_BALANCER_V1,
+            notes=f'Receive {receive_amt} BCoW-80QQQ-20WETH from a Balancer v1 pool',
+            address=balancer_qqq_weth_pool_token.evm_address,
+            extra_data={'deposit_events_num': 2},
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=166,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
```

Comment:
the order should be the other way. First deposit and then receive wrapped. THe logic is always out first and then in